src/notifications/ldn.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { LitDataset, Iri, Thing, IriString } from "../interfaces";
+import { dataset, DataFactory } from "../rdfjs";
+import { fetch } from "../fetcher";
+
+/**
+ * Perform partial inbox discovery (https://www.w3.org/TR/ldn/#discovery) by only checking
+ * resource content. If the inbox is only advsertised in the resource metadata, it won't be
+ * discovered.
+ *
+ * @param resource The IRI of the resource for which we are searching for the inbox
+ * @param dataset The dataset where the inbox may be found (typically fetched at the resource IRI)
+ */
+export function unstable_discoverInbox(
```

Comment:
Would this do an HTTP request? (I'd assume so, given that you're passing an IRI? If not, I'd maybe expect a `LitDataset` as parameter.) If so:

```suggestion
export async function unstable_discoverInbox(
```

(Also, for the documentation, maybe use the `Url | UrlString` types.)

And given that this is an external API, `UrlString` as return type :)

Reply:
I intended this function not to make an HTTP request, in order to make it possible to look for inboxes in a dataset that has already been fetched. Maybe the naming isn't good then, what about `_unstable_getInbox(...)` ? It's `unstable_fetchInbox` that does the actual fetching.

Reply:
And the return type is fixed in 611967efedfd15dbd52f59048e93882440246695